pkg/provider/aws/knowledgebase/edges.go

```diff
@@ -3,13 +3,45 @@ package knowledgebase
 import (
 	"fmt"
 	"reflect"
+	"strings"
 
 	"github.com/klothoplatform/klotho/pkg/core"
 	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base"
 	"github.com/klothoplatform/klotho/pkg/provider/aws/resources"
 )
 
 var AwsKB = knowledgebase.EdgeKB{
+	knowledgebase.Edge{Source: reflect.TypeOf(&resources.LambdaPermission{}), Destination: reflect.TypeOf(&resources.LambdaFunction{})}: knowledgebase.EdgeDetails{
+		Configure: func(source, dest core.Resource, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			permission := source.(*resources.LambdaPermission)
+			function := dest.(*resources.LambdaFunction)
+			if permission.Function != nil && permission.Function != function {
+				return fmt.Errorf("cannot configure edge %s -> %s, permission already tied to function %s", permission.Id(), function.Id(), permission.Function.Id())
+			}
+			permission.Function = function
+			return nil
+		},
+		ValidDestinations: []reflect.Type{reflect.TypeOf(&resources.LambdaFunction{})},
+	},
+	knowledgebase.Edge{Source: reflect.TypeOf(&resources.LambdaFunction{}), Destination: reflect.TypeOf(&resources.Subnet{})}: knowledgebase.EdgeDetails{
+		Configure: func(source, dest core.Resource, dag *core.ResourceGraph, data knowledgebase.EdgeData) error {
+			lambda := source.(*resources.LambdaFunction)
+			addVpcRole := true
+			for _, pol := range lambda.Role.AwsManagedPolicies {
+				if pol == "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole" {
+					addVpcRole = false
+				}
+			}
+			if addVpcRole {
+				lambda.Role.AwsManagedPolicies = append(lambda.Role.AwsManagedPolicies, "arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole")
+			}
```

Comment:
yeah it should, i could make that change now, but it may run into issues with any old code that exists

Reply:
I don't think anyone's working in that area. Not that you should change it in this PR per se, but it'd clean a lot up and be pretty low-hanging fruit.

Reply:
yeah let me see how much it would require to change

Reply:
i think i can knock it out now

Reply:
so i tried to do it and it makes the iac look awful since it doesnt align correctly then. Instead i think ill just create helper methods for these to make sure we dont have dedupes